src/com/itacademy/eshop/simulations/CustomerSimulation.java

```diff
@@ -29,21 +36,48 @@ private void addProductsToShoppingCart() {
          * User browses the shop and adds 4 products to the shopping cart.
          * One of them should be a book and other should be a shirt.
          */
-        throw new UnsupportedOperationException("addProductsToShoppingCart() is not implemented yet");
+
+        Product laptop = shop.findProductByName("Cooking Book");
+        shoppingCart.addProduct(laptop);
+
+        Product fridge = shop.findProductByName("Apple Watch");
+        shoppingCart.addProduct(fridge);
+
+        Product book = shop.findProductByName("Shirt");
+        shoppingCart.addProduct(book);
+
+        Product shirt = shop.findProductByName("Fridge");
+        shoppingCart.addProduct(shirt);
+
     }
 
     private void removeProductsFromShoppingCart() {
         /**
          * User removes shirt from the shopping cart.
          */
-        throw new UnsupportedOperationException("removeProductsFromShoppingCart() is not implemented yet");
+
+        shoppingCart.removeProductByName("Shirt");
+
     }
 
-private void leaveReviewsAndRatingsForProducts() {
+    private void leaveReviewsAndRatingsForProducts() {
         /**
          * User leaves 3 reviews and ratings for the products in the shopping cart.
          */
-        throw new UnsupportedOperationException("leaveReviewsAndRatingsForProducts() is not implemented yet");
+
+        List<Product> productList = shoppingCart.getProducts();
+
+        Author evaldas = new Author("Evaldas", "evaldas@gmail.com" );
+
+        Review review1 = new Review(evaldas, 5, "Amazing");
+        Review review2 = new Review(evaldas, 2, "Bad");
+        Review review3 = new Review(evaldas, 5, "Amazing");
+
```

Comment:
the variable above could be inlined to the method calls bellow, to avoid creating variables that are only used once